### PR TITLE
35 make batch report (Domain)

### LINKED ISSUE #35: make batch report (Domain)

### PR DESCRIPTION
**Pull request by dottore1**
closes #35

### CODE REVIEW

**Review by Wylardt — APPROVED** [review]

**Review by Lise4070 — APPROVED** [review]

**Review by Kaszz — APPROVED** [review]
🐸

**Review by Teyson — APPROVED** [review]

**Review by Teyson — CHANGES_REQUESTED** [review]
Rebase og få de sidste ting added :)

File: src/main/java/com/BrewMES/demo/BrewMesApplication.java [hunk 1]
```diff
@@ -1,15 +1,31 @@
...
+//			batch.addTemperature(LocalDateTime.now().plusSeconds(15), 15);
+//			batch.addTemperature(LocalDateTime.now().plusSeconds(20), 5);
+//			batch.addVibration(LocalDateTime.now(), 5);
+//			batch.addVibration(LocalDateTime.now().plusSeconds(10), 10);
+//			batch.addVibration(LocalDateTime.now().plusSeconds(55), 15);
+//			Report.generatePDF(batch);
 	}
-
-}
+}
```
Comment (Lawst96) [inline src/main/java/com/BrewMES/demo/BrewMesApplication.java:31]: Don't commit this, it's only used for testing and should not be merged.

**Review by Lawst96 — CHANGES_REQUESTED** [review]
Don't commit the test.pdf.

You should rename the package BrewMES to brewmes, so it matches the develop branch.

File: src/main/java/com/BrewMES/demo/BrewMesApplication.java (hunk 1 above)
Comment (dottore1) [reply src/main/java/com/BrewMES/demo/BrewMesApplication.java]: https://github.com/Semester3-03/Java-application/pull/105/commits/e0ad3f4b9d6c3a9b358224108e2016ac0f8abb9b

File: src/main/java/com/brewmes/demo/model/Report.java [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+        dataset.setValue(7, "state", "Execute");
+        dataset.setValue(10, "state", "Stopping");
+        dataset.setValue(2, "state", "Aborting");
+        dataset.setValue(2, "state", "Aborted");
+        dataset.setValue(2, "state", "Holding");
+        dataset.setValue(5, "state", "Resetting");
+        dataset.setValue(2, "state", "Completing");
+        dataset.setValue(2, "state", "Complete");
+        dataset.setValue(2, "state", "Deactivating");
+        dataset.setValue(8, "state", "Activating");
```
Comment (Lawst96) [inline src/main/java/com/brewmes/demo/model/Report.java:274]: Consider making "state" a variable, as to not write the same thing 17 times.

**Review by Lawst96 — CHANGES_REQUESTED** [review]

File: src/main/java/com/brewmes/demo/model/Report.java (hunk 2 above)
Comment (dottore1) [reply src/main/java/com/brewmes/demo/model/Report.java:229]: getting fixed when rebased :)

File: src/main/java/com/brewmes/demo/model/Machine.java [hunk 3]
```diff
@@ -253,6 +251,7 @@ public void setId(UUID id) {
         this.id = id;
     }
 
+
```
Comment (Lawst96) [inline src/main/java/com/brewmes/demo/model/Machine.java:254]:
```suggestion

```
Unnecessary white space addition

File: src/main/java/com/brewmes/demo/model/Machine.java [hunk 4]
```diff
@@ -264,7 +263,7 @@ public void setIp(String ip) {
     public Batch getCurrentBatch() {
         return currentBatch;
     }
-
+
```
Comment (Lawst96) [inline src/main/java/com/brewmes/demo/model/Machine.java:266]:
```suggestion
```
Unnecessary white space addition

**Review by Lawst96 — CHANGES_REQUESTED** [review]

File: src/main/java/com/brewmes/demo/BrewMesApplication.java [hunk 5]
```diff
@@ -1,22 +1,34 @@
...
+		batch.addTemperature(LocalDateTime.now().plusSeconds(20), 5);
+		Random rand = new Random();
+		for (int i = 0; i < 500; i++) {
+			batch.addVibration(LocalDateTime.now().plusSeconds(i+3),rand.nextInt(15));
+		}
+		batch.setTimeInState(0, 10);
+		batch.setTimeInState(1, 14);
+		batch.setTimeInState(2, 5);
+		batch.setTimeInState(4, 25);
+		Report.generatePDF(batch);
```
Comment (Teyson) [inline src/main/java/com/brewmes/demo/BrewMesApplication.java:32]: Alt ovenstående skal vel ikke på develop branchen?

**Review by Teyson — CHANGES_REQUESTED** [review]

File: src/main/java/com/brewmes/demo/BrewMesApplication.java (hunk 5 above)
Comment (dottore1) [reply src/main/java/com/brewmes/demo/BrewMesApplication.java]: https://github.com/Semester3-03/Java-application/pull/105/commits/f7fa0ea6fe5887cbf1b1dc94de3018fb9b0049e0

**Review by Lawst96 — APPROVED** [review]

**Review by Teyson — APPROVED** [review]

**Pull request merged**